tests/assembleCoaddTestUtils.py

```diff
@@ -91,36 +73,18 @@ def get(self, bbox=None, component=None, parameters=None):
             Either the exposure or its metadata, depending on the datasetType.
         """
         if component == 'psf':
-            return self.exposure.getPsf()
+            return self.inMemoryDataset.getPsf()
         elif component == 'visitInfo':
-            return self.exposure.getInfo().getVisitInfo()
+            return self.inMemoryDataset.getInfo().getVisitInfo()
+        bbox = None
         if parameters is not None:
-            if "bbox" in parameters:
-                bbox = parameters["bbox"]
-        exp = self.exposure.clone()
+            bbox = parameters.get("bbox")
+        exp = self.inMemoryDataset.clone()
```

Comment:
If this `clone()` is removed everything breaks because assembleMetadata gets no data sets but doesn't check it has none.

```python
tests/test_assembleCoadd.py:87: in runQuantum
    retStruct = self.run(mockSkyInfo, inputs.tempExpRefList, inputs.imageScalerList,
../../stack/lsst-scipipe-5.1.0/Darwin/utils/g8963c4024b+f0a1800e64/python/lsst/utils/timer.py:347: in timeMethod_wrapper
    res = func(self, *args, **keyArgs)
python/lsst/pipe/tasks/assembleCoadd.py:667: in run
    self.assembleMetadata(coaddExposure, tempExpRefList, weightList)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <test_assembleCoadd.MockAssembleCoaddTask object at 0x16c3de920>, coaddExposure = <lsst.afw.image.exposure.ExposureF object at 0x16af492f0>, tempExpRefList = [], weightList = []

    def assembleMetadata(self, coaddExposure, tempExpRefList, weightList):
        """Set the metadata for the coadd.
    
        This basic implementation sets the filter from the first input.
    
        Parameters
        ----------
        coaddExposure : `lsst.afw.image.Exposure`
            The target exposure for the coadd.
        tempExpRefList : `list`
            List of data references to tempExp.
        weightList : `list`
            List of weights.
    
        Raises
        ------
        AssertionError
            Raised if there is a length mismatch.
        """
        assert len(tempExpRefList) == len(weightList), "Length mismatch"
    
        # We load a single pixel of each coaddTempExp, because we just want to get at the metadata
        # (and we need more than just the PropertySet that contains the header), which is not possible
        # with the current butler (see #2777).
        bbox = geom.Box2I(coaddExposure.getBBox().getMin(), geom.Extent2I(1, 1))
    
        tempExpList = [tempExpRef.get(parameters={'bbox': bbox}) for tempExpRef in tempExpRefList]
    
        numCcds = sum(len(tempExp.getInfo().getCoaddInputs().ccds) for tempExp in tempExpList)
    
        # Set the coadd FilterLabel to the band of the first input exposure:
        # Coadds are calibrated, so the physical label is now meaningless.
>       coaddExposure.setFilter(afwImage.FilterLabel(tempExpList[0].getFilter().bandLabel))
E       IndexError: list index out of range

python/lsst/pipe/tasks/assembleCoadd.py:748: IndexError
```